scripts/mirror-builtin-actors.sh

```diff
@@ -0,0 +1,126 @@
+}
+
+# Function to send Slack alert with failed uploads
+send_slack_alert_with_failed() {
+    local failure_count=${#failed_uploads[@]}
+    local message="🚨 Filecoin Actors Mirror Update:\n🔥 Failed Uploads: $failure_count"
+
+    # Attach the log file with failed uploads
+    curl --form file=@"$FAILED_LOG" --form "initial_comment=$message" --form channels="$SLACK_CHANNEL" \
+         --header "Authorization: Bearer $SLACK_API_TOKEN" \
+         https://slack.com/api/files.upload
+}
+
+# Function to fetch and process releases
+fetch_and_process_releases() {
+    local page_url="$API_URL"
+
+    while [[ -n $page_url ]]; do
+        response=$(curl --silent --head "$page_url")
+        body=$(curl --silent "$page_url")
+        next_page_url=$(get_next_page_url "$response")
+
+        echo "$body" | jq --compact-output '.[]' | while read -r release; do
+            TAG_NAME=$(echo "$release" | jq --raw-output '.tag_name') || echo "Error: $release, could not get tag name"
+            PUBLISHED_DATE=$(echo "$release" | jq --raw-output '.published_at')
+            PUBLISHED_DATE_SEC=$(date --date="$PUBLISHED_DATE" +%s)
+
+            if echo "$TAG_NAME" | grep --extended-regexp '^v[0-9]+\.[0-9]+\.[0-9]+.*$' && [[ "$PUBLISHED_DATE_SEC" -ge "$THREE_YEARS_AGO" ]]; then
+                mkdir --parents "$BASE_FOLDER/$TAG_NAME"
+            fi
+        done
+
+        page_url="$next_page_url"
+    done
+}
+
+fetch_and_process_releases
+
+declare -a failed_uploads
+failed_uploads=()
+
+: 'Loop through each version directory to process and upload assets'
+while IFS= read -r version_dir; do
+    TAG_NAME=${version_dir#"$BASE_FOLDER"/}
+    VERSION_DIR="$version_dir"
+    if [ -d "$VERSION_DIR" ]; then
+        echo "Entering directory: $VERSION_DIR"
+
+        tag_url="$API_URL/tags/$TAG_NAME"
+        release=$(curl --silent "$tag_url")
+
+        # Check if the assets array is not null
+        if [[ $(echo "$release" | jq '.assets') != "null" ]]; then
+            ASSETS=$(echo "$release" | jq --compact-output '.assets[]')
+
+            pushd "$VERSION_DIR" > /dev/null
+            echo "Processing assets for $TAG_NAME..."
+
+            echo "$ASSETS" | while IFS= read -r asset; do
+                DOWNLOAD_URL=$(echo "$asset" | jq --raw-output '.browser_download_url')
+                FILE_NAME=$(echo "$asset" | jq --raw-output '.name')
+
+                echo "Checking asset: $FILE_NAME"
+                if [ ! -f "$FILE_NAME" ]; then
+                    echo "Downloading $FILE_NAME..."
+                    curl --location --silent --show-error --output "$FILE_NAME" "$DOWNLOAD_URL" || echo "Failed to download $FILE_NAME"
+
+                fi
+
+                echo "Checking $FILE_NAME against S3 version..."
+                TEMP_S3_DIR=$(mktemp --directory)
+                s3cmd get --no-progress "s3://$BUCKET_NAME/$TAG_NAME/$FILE_NAME" "$TEMP_S3_DIR/$FILE_NAME" || true
```

Comment:
why download the file here? these assets should be kept immutable

Reply:
we're downloading it so that we can compare the s3 version against the one in Github. If there's any difference or missing actor, upload the downloaded actor from Github